docs/source/reference/packages.rst

```diff
@@ -162,3 +162,47 @@ package will then be located as follows:
 In each case, if more than one version satisfies the constraint, it will choose
 the one with the highest version number.
 If package versions are omitted in directory names, they are treated as the version ``0``.
+
+Support file install directories
+================================
+
+As noted above, packages can be installed globally or locally to be used as dependencies in
+other projects. In either case, there are two directories within a package's install root that
+Idris 2 treats specially when the package is depended upon by some other Idris package.
+
+Those directories are ``lib`` and ``data``.
+
+Library files
+-------------
+
+Idris will look for library files at ``pkgname-<version>/lib``. Library files are those that
+will need to be around whenever an executable that depends on your library is run. That is,
+these files are not built into the executable, but rather linked against or otherwise referred
+to during building and then loaded in at runtime. A common example of this type of file is a
+shared object file (these commonly have the extension ``.so`` on Linux systems and ``.dylib`` on Mac OS
+systems).
+
+When building with the Scheme or Refc backends, you may want your package to use Idris 2's ``C`` FFI
+and rely on a ``C`` support file. The FFI interface is documented elsewhere so we won't go into that
+here. Once you do have a support file written in ``C``, you can build an ``so`` file using your
+package's post-build hook. Then, install (copy) that ``so`` file into the ``lib`` subfolder where
+Idris has installed your package in your post-install hook. When an executable depends on your package,
+Idris will copy shared object files from the ``lib`` directory into the build folder for the exectuable.
+
+Data files
+----------
+
+Idris will look for data files at ``pkgname-<version>/data``. Data files are a bit open-ended
```

Comment:
I think it would be good to make it clear this is refering to the global idris 2 install directory
(ie something like: idris will look for data files in idris's install directory, at <path>, where path is something along the lines of `$(idris2 --libdir)/...` (I can't remember the exact path right now))

Reply:
I think that information is covered by the preceding section ("Where does Idris look for packages?") pretty adequately so I attempted to make it clear I was referring to that directory at the top of the new section by saying 
>  noted above, packages can be installed globally or locally [...]. In either case, there are two directories within a package's install root that Idris 2 treats specially [...].

Do you think I should be clearer on that? Either way, it would not quite be accurate to say that the `pkgname-<version>/data` directory was under the global install location because it should work the same way when a dependency lives in the local `depends` directory.

Reply:
Good point, I think would be useful to say where/how to figure out where that global directory is.
Although if you are planning to auto-install support files, this might not be so important

Reply:
Can't hurt, I'll add more detail to the top of the section (since the information applies to locating both the "data" and "lib" directories.

Reply:
Let me know if you think the new text adds enough clarity.

Reply:
Looks good!